saga-spring/src/main/java/io/servicecomb/saga/spring/SagaController.java

```diff
@@ -91,11 +103,104 @@ public SagaController(SagaExecutionComponent sagaExecutionComponent, SagaEventRe
+      @RequestParam(name = "pageIndex") String pageIndex,
+      @RequestParam(name = "pageSize") String pageSize,
+      @RequestParam(name = "startTime") String startTime,
+      @RequestParam(name = "endTime") String endTime) {
+    if (isRequestParamValid(pageIndex, pageSize, startTime, endTime)) {
+      List<SagaRequest> requests = new ArrayList<>();
+      Page<SagaEventEntity> startEvents = repo.findByTypeAndCreationTimeBetweenOrderByIdDesc(
+          SagaStartedEvent.class.getSimpleName(),
+          new Date(Long.parseLong(startTime)), new Date(Long.parseLong(endTime)),
+          new PageRequest(Integer.parseInt(pageIndex), Integer.parseInt(pageSize)));
+      for (SagaEventEntity event : startEvents) {
+        SagaEventEntity endEvent = repo
+            .findFirstByTypeAndSagaId(SagaEndedEvent.class.getSimpleName(), event.sagaId());
+        SagaEventEntity abortedEvent = repo
+            .findFirstByTypeAndSagaId(TransactionAbortedEvent.class.getSimpleName(), event.sagaId());
+
+        requests.add(new SagaRequest(
+            event.id(),
+            event.creationTime(),
+            endEvent == null ? 0 : endEvent.creationTime(),
+            endEvent == null ? "Running" : abortedEvent == null ? "OK" : "Failed"));
+      }
+      return ResponseEntity.ok(
+          new SagaRequestQueryResult(Integer.parseInt(pageIndex), Integer.parseInt(pageSize),
+              startEvents.getTotalPages(), requests));
+    } else {
+      throw new InvocationException(BAD_REQUEST, "illegal request content");
+    }
+  }
+
+  private boolean isRequestParamValid(String pageIndex, String pageSize, String startTime, String endTime) {
+    return Integer.parseInt(pageIndex) >= 0 && Integer.parseInt(pageSize) > 0
+        && Long.parseLong(startTime) <= Long.parseLong(endTime);
+  }
+
+  @ApiResponses({
+      @ApiResponse(code = 200, response = String.class, message = "success"),
+      @ApiResponse(code = 400, response = String.class, message = "illegal request content"),
+  })
+  @RequestMapping(value = "requests/{sagaId}", method = GET)
+  public ResponseEntity<List<SagaEventVo>> queryRequest(@PathVariable String sagaId) {
+    Iterable<SagaEventEntity> entities = repo.findBySagaId(sagaId);
+    List<SagaEventVo> events = new ArrayList<>();
+    entities
+        .forEach(e -> events.add(new SagaEventVo(e.id(), e.sagaId(), e.creationTime(), e.type(), e.contentJson())));
+    return ResponseEntity.ok(events);
+  }
+
   @JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
   private static class SagaEventVo extends SagaEventEntity {
 
     private SagaEventVo(long id, String sagaId, long creationTime, String type, String contentJson) {
       super(id, sagaId, creationTime, type, contentJson);
     }
   }
+
+  @JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
+  protected static class SagaRequestQueryResult {
```

Comment:
it seems these two classes are only used for serialization not deserialization. if so, there's no need for no param constructor.

Reply:
Done